jax/experimental/mosaic/gpu/core.py

```diff
@@ -798,6 +799,8 @@ def _shape_to_ref_ty(shape: jax.ShapeDtypeStruct) -> ir.MemRefType:
   module = ir.Module.create()
   attrs = module.operation.attributes
   attrs["sym_name"] = ir.StringAttr.get(module_name)
+  if kernel_name is None:
+    kernel_name = body.__name__
```

Comment:
You can use `util.fun_name` here, but note that `body` doesn't necessarily have a name in the general case or it could be something like `"<lambda>"`.

Reply:
`util.fun_name` didn't work out here because its default uses characters that LLVM rejects in a symbol name:
https://github.com/jax-ml/jax/blob/a212a29dc654986ba2d85255a71797490c631c2a/jax/_src/util.py#L378-L379

but I used it as inspiration.